### DISCUSSION TITLE
signInAnonymously Confusion with JS SDK

### DISCUSSION THREAD

**Discussion by DevDianDankie** [capawesome-team/capacitor-firebase#388] [Q&A] [answered]
Hi there.

I am struggling with signInAnonymously due to using JS SDK.

I notice the general flow is to sign in to the native layer but then also sign in to the web layer so I can use the firestore. This is working for all other OAuth providers via signInWithCredential after singing in natively.

How am I suppose to do it for signInAnonymously? Is it even possible 👀 ?

Is there something I can do with the returned SignInResult to sign in on the web layer? I hope so 😅 ...

**Comment by robingenz** [comment]
You don't need the native SDK for "signInAnonymously" if you also want to use the Firebase JS SDK. Just use the Firebase JS SDK directly to sign in anonymously.

**Comment by DevDianDankie** [comment]
mmm I tried that initially. But then when the time comes to link with google or apple or email It gives an error that there is no one signed in 😓.

I use this (which might also not be correct):

`
  async linkWithGoogle() {
      const result = await FirebaseAuthentication.linkWithGoogle();
      const credential = GoogleAuthProvider.credential(result.credential?.idToken);
      if(Capacitor.isNativePlatform()){
        await linkWithCredential(this.fAuth.currentUser, credential); 
      }
      
      const googleUser = await signInWithCredential(this.fAuth, credential);
      return googleUser
    
  }

  async linkWithApple(){
    const result = await FirebaseAuthentication.linkWithApple();
    const provider = new OAuthProvider('apple.com');
    const credential = provider.credential({
      idToken: result.credential?.idToken,
      rawNonce: result.credential?.nonce,
      });
      if(Capacitor.isNativePlatform()){
        await linkWithCredential(this.fAuth.currentUser, credential); 
    }
    const appleUser = await signInWithCredential(this.fAuth, credential);
    return appleUser
  }

  async linkWithEmail (email:string,password:string,displayName:string){
   await FirebaseAuthentication.linkWithEmailAndPassword({
      email,
      password
    })
    const credential = EmailAuthProvider.credential(email,password);
   
    if(Capacitor.isNativePlatform()){
      await linkWithCredential(this.fAuth.currentUser, credential); 
  }
    const user = await signInWithCredential(this.fAuth, credential);
    await updateProfile(user.user,{displayName});
    return user
  }
`

**Reply by robingenz** [reply, marked as answer]
Please take a look at this "linkWithGoogle()" example: https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/authentication/docs/firebase-js-sdk.md

**Reply by DevDianDankie** [reply]
ah ok. so I need to use sign in methods instead. lemme try this. Will report back.

**Reply by DevDianDankie** [reply]
Thank you! indeed seems to work!
I am using these now:

async linkWithGoogle() {
      const result = await FirebaseAuthentication.signInWithGoogle();
      const credential = GoogleAuthProvider.credential(result.credential?.idToken);
      const googleUserCred= await linkWithCredential(this.afAuth.currentUser, credential); 
      await updateProfile(googleUserCred.user,{displayName:googleUserCred.user.displayName,photoURL:googleUserCred.user.photoURL});
      await this.dbService.initUser(googleUserCred.user);
      return googleUserCred

  }

  async linkWithApple(){
    const result = await FirebaseAuthentication.signInWithApple({
        skipNativeAuth: true,
      });
    const provider = new OAuthProvider('apple.com');
    const credential = provider.credential({
      idToken: result.credential?.idToken,
      rawNonce: result.credential?.nonce,
      });
      const appleUserCred = await linkWithCredential(this.afAuth.currentUser, credential); 
      await updateProfile(appleUserCred.user,{displayName:appleUserCred.user.displayName,photoURL:appleUserCred.user.photoURL});
      await this.dbService.initUser(appleUserCred.user);
    return appleUserCred
  }

  async linkWithEmail (email:string,password:string,displayName:string){
    const credential = EmailAuthProvider.credential(email,password);
    const userCred = await linkWithCredential(this.afAuth.currentUser, credential); 
    await updateProfile(userCred.user,{displayName});
    await this.dbService.initUser(userCred.user);
    return userCred
  }
